### DISCUSSION TITLE
CC2652P2 Router output power

### DISCUSSION THREAD

**Discussion by nomis** [tube0013/tube_gateways#157] [Q&A]
The co-ordinator has "Output power up to +20 dBm" but the router based on the same CC2652P2 only has "Output power up to +5 dBm (could be increased in the future with FW update)".

I'm having problems with SONOFF S26R2ZB devices having awful RSSI readings because they can only transmit at 5 dBm and most of them are located far away from the co-ordinator and each other.

I want to improve the situation by putting dedicated routers in specific locations, but that won't help if the routers can only transmit at 5 dBm too because they won't have a good signal to the co-ordinator without using an excessive number of routers.

What I want is a network of "Co-ordinator ⇐(20 dBm)⇒ Routers ⇐(5 dBm)⇒ End Devices" but to do that the routers need to be able to transmit at 20 dBm too.

**Comment by tube0013** [comment]
Current router shipping fw and also in repository supports configurable tx power up to 20. It is based on the most recent 20221102 z-stack router fw.

**Reply by nomis** [reply]
Is there a way to get the RSSI readings for all of the router's neighbours?

There's a ZBOSS API that I can use for this on an ESP32 router.

It looks like Z-Stack [has support for the RSSI Location cluster](https://software-dl.ti.com/simplelink/esd/simplelink_cc13x2_26x2_sdk/3.10.00.53/exports/docs/zigbee/doxygen/zigbee/html/group___z_c_l___g_e_n_e_r_a_l___r_s_s_i___l_o_c_a_t_i_o_n___c_l_u_s_t_e_r.html0) but I don't know if the router supports that? (or a subset of it, because I just want RSSI readings and don't need the X,Y,Z location calculation)
